src/Raven.Studio/typescript/common/shell/databasesManager.ts

```diff
@@ -62,22 +63,29 @@ class databasesManager {
             });
     }
 
-    activateBasedOnCurrentUrl(): JQueryPromise<canActivateResultDto> | boolean {
-        const dbUrl = appUrl.getDatabaseNameFromUrl();
-        if (dbUrl) {
-            const db = this.getDatabaseByName(dbUrl);
-            return this.activateIfDifferent(db, dbUrl);
+    activateBasedOnCurrentUrl(dbName: string, isViewAllowed: boolean): JQueryPromise<canActivateResultDto> | boolean {
+        if (dbName) {
+            const db = this.getDatabaseByName(dbName);
+            return this.activateIfDifferent(db, dbName, isViewAllowed);
         }
         return true;
     }
 
-    private activateIfDifferent(db: database, dbName: string): JQueryPromise<canActivateResultDto> {
+    private activateIfDifferent(db: database, dbName: string, isViewAllowed: boolean): JQueryPromise<canActivateResultDto> {
         const task = $.Deferred<canActivateResultDto>();
+        const databasesListView = appUrl.forDatabases();
 
         const incomingDatabaseName = db ? db.name : undefined;
 
         this.initialized.done(() => {
             const currentActiveDatabase = this.activeDatabaseTracker.database();
+
+            if (!isViewAllowed) {
+                messagePublisher.reportError("Redirecting. Invalid access level.");
```

Comment:
==> Forbidden access.

Reply:
Changing to:
`Redirecting. Access is forbidden.`

ok ?

Reply:
I'd start with cause then with reaction. 
Access is forbidden. Redirecting to databases view. 

But in this case I'm not sure second part is needed, since the most important thing is: You don't have an access. 
How we "heal" this state is not so important.

Reply:
ok - done